### PR TITLE
Simple non-antlr AT parser

### PR DESCRIPTION
**Pull request by lukebemish**
Antlr, unfortunately, is Not Great. At this point, it's beginning to consistently be badly-supported by gradle and IDEs alike; furthermore, it's not necessary at *all* for this use case. AT files are quite easy to parse; here, I implement a small, regex-free AT file parser that as far as I can tell is consistent with the old behavior of AT files. While it seems to run correctly on all the tests, if anyone finds an issue with my implementation, please let me know.

### CODE REVIEW

**Comment by FiniteReality** [comment]
I feel that we should keep the ANTLR grammars around, though not in `/src`, likely a `/extras` folder or some such. This is because it may be useful to other people writing AT parsers as a reference (e.g. for IDE integration), even if it's not a 1:1 match with how they're parsed actively.

**Comment by lukebemish** [comment]
That's not a bad suggestion; added them in a `extra` directory.

File: src/main/java/net/neoforged/accesstransformer/AccessTransformer.java [hunk 1]
```diff
@@ -1,99 +1,121 @@
...
+        } else if (target instanceof Target.ClassTarget classTarget) {
+            return new ClassAccessTransformer(classTarget, transformation);
+        } else if (target instanceof Target.InnerClassTarget innerClassTarget) {
+            return new InnerClassAccessTransformer(innerClassTarget, transformation);
+        } else if (target instanceof Target.WildcardFieldTarget wildcardFieldTarget) {
+            return new WildcardAccessTransformer(wildcardFieldTarget, transformation);
+        } else if (target instanceof Target.WildcardMethodTarget wildcardMethodTarget) {
+            return new WildcardAccessTransformer(wildcardMethodTarget, transformation);
+        } else {
+            // It's sealed, this shouldn't happen - we'll want to make this whole thing a switch with J21
```
Comment (Matyrobbrt) [inline src/main/java/net/neoforged/accesstransformer/AccessTransformer.java:40]: You can bump to Java 21 if you want.

File: src/main/java/net/neoforged/accesstransformer/AccessTransformerEngineImpl.java [hunk 2]
```diff
@@ -11,13 +10,10 @@
...
 import java.io.IOException;
+import java.io.Reader;
 import java.net.URISyntaxException;
 import java.nio.file.Path;
-import java.util.HashSet;
-import java.util.Map;
-import java.util.Set;
-import java.util.Spliterator;
-import java.util.Spliterators;
+import java.util.*;
```
Comment (Matyrobbrt) [inline src/main/java/net/neoforged/accesstransformer/AccessTransformerEngineImpl.java:16]: No wildcard imports.

**Review by Matyrobbrt — CHANGES_REQUESTED** [review]

File: src/main/java/net/neoforged/accesstransformer/AccessTransformerEngineImpl.java (hunk 2 above)
Comment (lukebemish) [reply src/main/java/net/neoforged/accesstransformer/AccessTransformerEngineImpl.java:16]: AT already has a good number. Can we set up spotless in another PR maybe?

Comment (lukebemish) [reply src/main/java/net/neoforged/accesstransformer/AccessTransformerEngineImpl.java]: Fixed them here instead

**Review by Matyrobbrt — APPROVED** [review]

**Pull request merged**